engine/src/main/java/org/camunda/bpm/engine/impl/util/ParseUtil.java

```diff
@@ -116,4 +116,18 @@ public static ProcessEngineDetails parseProcessEngineVersion(String packageImple
 
     return new ProcessEngineDetails(version, edition);
   }
+
+  public static String parseServerVendor(String applicationServerInfo) {
+    String serverVendor = null;
+    Pattern pattern = Pattern.compile("[\\sA-Za-z]+");
+    Matcher matcher = pattern.matcher(applicationServerInfo);
+    if (matcher.find()) {
+      serverVendor = matcher.group(0).trim();
+    }
+    if (serverVendor.contains("WildFly")) {
+      return "WildFly";
+    }
+    return serverVendor;
```

Comment:
A proposal to be super sure that an exception is **never** thrown:

```suggestion
    String serverVendor = "";

    Pattern pattern = Pattern.compile("[\\sA-Za-z]+");
    Matcher matcher = pattern.matcher(applicationServerInfo);
    if (matcher.find()) {
      try {
        serverVendor = matcher.group();
      } catch (IllegalStateException ignored) {
      }

      serverVendor = serverVendor.trim();

      if (serverVendor.contains("WildFly")) {
        return "WildFly";
      }
    }

    return serverVendor;
```